### PR TITLE
Implement guts in cells

### LINKED ISSUE #2: Implement guts in Cell
### Expected Behavior

When content named guts.xml  is required from a cell a new byte[] containing names and sizes of the cell content is returned
### Actual Behavior

a content named guts.xml is returned if existing or a empty string

### Mention any other details that might be useful

fileCell guts is ramcell, too

### PR DESCRIPTION
**Pull request by jaanpeeter**
### Please check if the PR fulfills these requirements
- [x] I made sure that my code builds
- [x] I merged the master into this branch before pushing
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce?** (check one with "x")
- [x] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

### What is the current behavior? (You can also link to an open issue here)

#2 
### What is the new behavior?

closes #2 
fix: fixes RamCell name in RamCom
### Does this PR introduce a breaking change? (check one with "x")
- [ ] Yes
- [x] No

###  Other information

### CODE REVIEW

File: src/Xive/Cell/FileCell.cs [hunk 1]
```diff
@@ -68,7 +70,33 @@ private FileCell(IScalar<string> path)
...
         {
             byte[] result = new byte[0];
-            if (File.Exists(this.path.Value()))
+
+            if (this.path.Value().Equals("_guts.xml"))
+            {
+                var patch = new Directives().Add("items");
+
+
+                foreach (var file in Directory.GetFiles(
```
Comment (Meerownymous) [inline src/Xive/Cell/FileCell.cs]: Like you already mentioned: Cell is the wrong place for guts aggregation.

File: src/Xive/Comb/FileComb.cs [hunk 2]
```diff
@@ -62,7 +64,40 @@ public FileComb(string root, string name, Func<ICell, ICell> cellWrap, Func<IXoc
...
+                var patch = new Directives().Add("items");
+
+                foreach (var file in Directory.GetFiles(
+                        Path.GetDirectoryName(this.comb.Value().Name()),
+                        "*.*",
+                        SearchOption.AllDirectories))
+                {
+                    patch.Add("item")
+                        .Add("name")
+                        .Set(Path.GetFileName( file)) //file.Substring((this.name + "/").Length))
```
Comment (Meerownymous) [inline src/Xive/Comb/FileComb.cs]: Remove comment (and unneeded space in brackets)

File: src/Xive/Comb/RamComb.cs [hunk 3]
```diff
@@ -98,9 +100,38 @@ public IXocument Xocument(string name)
             return this.xocumentWrap(new CellXocument(Cell(name), name));
         }
 
-        public ICell Cell(string name)
+        public ICell Cell(string cell)
```
Comment (Meerownymous) [inline src/Xive/Comb/RamComb.cs]: Change back to name.
Method name already tells us that we get a cell.
Variable reflects what to define to get it.

File: src/Xive/Comb/RamComb.cs [hunk 4]
```diff
@@ -98,9 +100,38 @@ public IXocument Xocument(string name)
...
+            ICell result;
+            if (cell.Equals("_guts.xml"))
+            {
+                var patch = new Directives().Add("items");
+
+                foreach (var name in this.cellMemory.Keys)
+                {
+                    patch.Add("item")
+                        .Add("name")
+                        .Set(name) //file.Substring((this.name + "/").Length))
```
Comment (Meerownymous) [inline src/Xive/Comb/RamComb.cs]: We want guts to tell us the path from here - the comb. So you must take the substring, cutting off the comb name from the full cell name.

The dictionary will alwas contain a full name, think of it like a file path. Same concept.

File: src/Xive/Cell/FileCell.cs (hunk 1 above)
Comment (jaanpeeter) [reply src/Xive/Cell/FileCell.cs]: Looks like I do not know how to undo things via git!

File: src/Xive/Comb/FileComb.cs (hunk 2 above)
Comment (jaanpeeter) [reply src/Xive/Comb/FileComb.cs]: comment removed - which unneeded space?

Comment (koeeenig) [reply src/Xive/Comb/FileComb.cs]: the leading space in front of `file`

Comment (jaanpeeter) [reply src/Xive/Comb/FileComb.cs]: thanks, sorry was focussed on the foreach brackets

**Pull request merged**